packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts

```diff
@@ -240,18 +245,24 @@ export class PersonalizeMiddleware extends MiddlewareBase {
     const rewritePath = getPersonalizedRewrite(basePath, { variantId });
     // Note an absolute URL is required: https://nextjs.org/docs/messages/middleware-relative-urls
     const rewriteUrl = req.nextUrl.clone();
+
     rewriteUrl.pathname = rewritePath;
-    response = NextResponse.rewrite(rewriteUrl);
+    response = NextResponse.rewrite(rewriteUrl, response);
+
+    const setCookieHeader = response.headers.get('set-cookie');
 
     // Disable preflight caching to force revalidation on client-side navigation (personalization may be influenced)
     // See https://github.com/vercel/next.js/issues/32727
     response.headers.set('x-middleware-cache', 'no-cache');
-    // Share rewrite path with following executed middlewares
+    // Share rewrite path with following executed middleware
     response.headers.set('x-sc-rewrite', rewritePath);
-
     // Share site name with the following executed middlewares
     response.cookies.set(this.SITE_SYMBOL, site.name);
 
+    if (setCookieHeader) {
+      response.headers.set('set-cookie', setCookieHeader);
```

Comment:
This looks a bit different than I was expecting. Wouldn't this override the preceding `response.cookies.set(this.SITE_SYMBOL, site.name);`? 

Might be worth explaining what's going on here with the get/set of 'set-cookie' header (and why we can't use `cookies.set`).

Reply:
makes sense.
Refactored.